bot/exts/evergreen/profile_pic_modification/pfp_modify.py

```diff
@@ -0,0 +1,241 @@
+        async with ctx.typing():
+            file = await in_executor(
+                PfpEffects.apply_effect,
+                image_bytes,
+                PfpEffects.pridify_effect,
+                pixels,
+                flag
+            )
+
+            embed = discord.Embed(
+                name="Your Lovely Pride Avatar!",
+                description=f"Here is your lovely avatar, surrounded by\n a beautiful {option} flag. Enjoy :D"
+            )
+            embed.set_image(url="attachment://modified_avatar.png")
+            embed.set_footer(text=f"Made by {ctx.author.display_name}.", icon_url=ctx.author.avatar_url)
+            await ctx.send(file=file, embed=embed)
+
+    @pfp_modify.group(
+        aliases=["avatarpride", "pridepfp", "prideprofile"],
+        root_aliases=("prideavatar", "avatarpride", "pridepfp", "prideprofile"),
+        invoke_without_command=True
+    )
+    async def prideavatar(self, ctx: commands.Context, option: str = "lgbt", pixels: int = 64) -> None:
+        """
+        This surrounds an avatar with a border of a specified LGBT flag.
+
+        This defaults to the LGBT rainbow flag if none is given.
+        The amount of pixels can be given which determines the thickness of the flag border.
+        This has a maximum of 512px and defaults to a 64px border.
+        The full image is 1024x1024.
+        """
+        option = option.lower()
+        pixels = max(0, min(512, pixels))
+        flag = self.GENDER_OPTIONS.get(option)
+        if flag is None:
+            await ctx.send("I don't have that flag!")
+            return
+
+        async with ctx.typing():
+            image_bytes = await ctx.author.avatar_url.read()
+            await self.send_pride_image(ctx, image_bytes, pixels, flag, option)
+
+    @prideavatar.command()
+    async def image(self, ctx: commands.Context, url: str, option: str = "lgbt", pixels: int = 64) -> None:
+        """
+        This surrounds the image specified by the URL with a border of a specified LGBT flag.
+
+        This defaults to the LGBT rainbow flag if none is given.
+        The amount of pixels can be given which determines the thickness of the flag border.
+        This has a maximum of 512px and defaults to a 64px border.
+        The full image is 1024x1024.
+        """
+        option = option.lower()
+        pixels = max(0, min(512, pixels))
+        flag = self.GENDER_OPTIONS.get(option)
+        if flag is None:
+            await ctx.send("I don't have that flag!")
+            return
+
+        async with ctx.typing():
+            async with self.bot.http_session as session:
+                try:
+                    response = await session.get(url)
+                except client_exceptions.ClientConnectorError:
+                    await ctx.send("Cannot connect to provided URL!")
+                    return
+                except client_exceptions.InvalidURL:
+                    await ctx.send("Invalid URL!")
```

Comment:
We can probably raise a `BadArgument` error here, but am fine both the cases, the current one and raising a error (letting the error handler handle it).